user_modules/shared_testcases/commonFunctions.lua

```diff
@@ -51,7 +54,75 @@ function commonFunctions:isMediaApp()
   end
 
   return isMedia
+end
+
+function commonFunctions:waitForSDLStart(test)
+  local step = 100
+  local hmiPort = config.hmiPort
+  local event = events.Event()
+  event.matches = function(self, e) return self == e end
+  local function raise_event()
+    --check that sdl ports are open then raise else RUN after 100 ms
+    local output = os.execute ("netstat -vatn  | grep " .. hmiPort .. " | grep LISTEN")
+    if (output) then
+      RAISE_EVENT(event, event)
+    else 
+      RUN_AFTER(raise_event, step)
+    end
+  end
+  RUN_AFTER(raise_event, step)
+  local ret = expectations.Expectation("Wait for SDL start", test.mobileConnection)
+  ret.event = event
+  event_dispatcher:AddEvent(test.mobileConnection, ret.event, ret)
+  test:AddExpectation(ret)
+  return ret
+end
+
+function commonFunctions:createMultipleExpectationsWaiter(test, name)
+  assert(name)
+  exp_waiter = {}
+  exp_waiter.expectation_list = {}
+  function exp_waiter:CheckStatus()
+     if #exp_waiter.expectation_list == 0 and not exp_waiter.expectation.status then 
+      exp_waiter.expectation.status = SUCCESS
+      event_dispatcher:RaiseEvent(test.mobileConnection, exp_waiter.event)
+      return true
+     end
+     return false 
+  end
+
+  function exp_waiter:AddExpectation(exp)
+    table.insert(exp_waiter.expectation_list, exp)
+    exp:Do(function()
+      exp_waiter:RemoveExpectation(exp)
+      exp_waiter:CheckStatus()
+    end)
+  end
+
+  function exp_waiter:RemoveExpectation(exp)
+    local function AnIndexOf(t,val)
```

Comment:
@LuxoftAKutsan No, I did not find function for this purpose in our commom modules